sars2seq/translate.py

```diff
@@ -175,10 +175,246 @@ def checkSpikeInsertions(accession, seq):
+            return seq[:264] + seq[265:]
+        elif seq.find('SALLSDLQGTIT', 870, 890) > -1:   # -880D -881L -882Q
+            return seq[:879] + seq[882:]
+        elif seq.find('TEKSKAENIIR', 93, 103) > -1:  # -99K -100A -101E
+            return seq[:98] + seq[101:]
+        elif seq.find('MFVFFFVL', 0, 10) > -1:   # -4F L6F
+            return seq[:3] + seq[4:]
+        elif seq.find('TWLLGSMHIS', 55, 80) > -1:  # -65L F66L H67G A68S I69M
+            return seq[:64] + seq[65:]
+        elif seq.find('LHRSSHQNLTP', 239, 254) > -1:  # -249H -250Q -251N
+            return seq[:248] + seq[251:]
+        elif seq.find('NLVRAKKNDLP', 210, 220) > -1:  # -215A -216K -217K -218N
+            return seq[:214] + seq[218:]
+        elif seq.find('VSQPFFFMD', 167, 182) > -1:  # -175F
+            return seq[:174] + seq[175:]
+        elif seq.find('NLVRKLGPDLPQG', 207, 221) > -1:  # -215K -216L -217G
+            return seq[:214] + seq[218:]                # -218P
+        elif seq.find('LHAPPATV', 515, 525) > -1:   # -521P
+            return seq[:520] + seq[521:]
+        elif seq.find('LLHAAPPATV', 515, 525) > -1:  # -521A -522P
+            return seq[:520] + seq[522:]
+        elif seq.find('LHAHPATV', 515, 525) > -1:   # -521H
+            return seq[:520] + seq[521:]
+        elif seq.find('LLHAPPPATV', 515, 525) > -1:  # -521P -522P
+            return seq[:520] + seq[522:]
+        elif seq.find('PGDSSSS', 248, 258) > -1:  # -254S
+            return seq[:253] + seq[254:]
+        elif seq.find('AWNRKRISKRIS', 350, 360) > -1:  # -356K -357R -358I
+            return seq[:355] + seq[359:]               # -359S
+        elif seq.find('LIGAAEH', 645, 660) > -1:  # -653A
+            return seq[:652] + seq[653]
+        elif seq.find('ALHRDSWGSY', 240, 250) > -1:  # -247D -248S -249W -250G
+            return seq[:246] + seq[250:]
+        elif seq.find('CASYQTQTQT', 667, 680) > -1:  # -675Q -676T
+            return seq[:674] + seq[676:]
+        elif seq.find('GSCCKFKFD', 1250, 1260) > -1:  # -1255K -1256F
+            return seq[:1254] + seq[1256]
+        elif seq.find('MLVFFF', 0, 10) > -1:  # -1M M2L
+            return seq[:3] + seq[4:]
+        elif seq.find('SQCATLRINLT', 10, 20) > -1:  # -17T -18L -19R -20I
+            return seq[:16] + seq[20:]
+        elif seq.find('FNDGVCVY', 84, 95) > -1:  # -90V -91C
+            return seq[:89] + seq[91:]
+        elif seq.find('NPVLPLPFN', 78, 89) > -1:  # -85P -86L
+            return seq[:84] + seq[86:]
+        elif seq.find('VLPFNVND', 79, 91) > -1:  # -87N -88V
+            return seq[:86] + seq[88:]
+        elif seq.find('VLPFNDDDG', 80, 92) > -1:  # -88D -89D
+            return seq[:87] + seq[:89]
+        elif seq.find('FDNPVLPLPF', 75, 90) > -1:  # -84L -85P
+            return seq[:83] + seq[85:]
+        elif seq.find('LHRSSSLTYLT', 240, 255) > -1:  # -248S -249S -250L -251T
+            return seq[:247] + seq[251:]
```

Comment:
This is all extremely repetitive, as I'm sure you know! :-). And code like this is always error prone.

You could instead do this:

```python
CORRECTIONS = (
    ('LHRSSSLTYLT', 240, 255, 247, 251), # -248S -249S -250L -251T
    ('KVCEFQFQ', 125, 137, 132, 134),  # -133F -134Q
)

for target, findStart, findStop, sliceStart, sliceStop in CORRECTIONS:
    if seq.find(target, findStart, findStop) > -1:
        return seq[:sliceStart] + seq[sliceStop:]

raise TranslatedSequenceLengthError(f'...')
```

The code could `assert findStart < findStop` but it would be better to write a test for that, importing `CORRECTIONS`. You could then also test that no two amino acid sequences are the same, which it would be important to know.

I can do the edits for you easily / quickly in emacs if you like.

Reply:
A better test would be `self.assertTrue(findStop - findStart > len(target))`.

Reply:
But I guess that's ok....

Reply:
OK, changed and added tests.